### ISSUE TITLE
Problem with extending PageTranslation

### ISSUE THREAD

**Issue by antonioperic** [BitBagCommerce/SyliusCmsPlugin#164]
Hi, 
seems there is some issue when PageTranslation is extended. I wanted to add $subttitle, this is my flow

```
sylius_resource:

    resources:
        bitbag_sylius_cms_plugin.page:
            classes:
                model: AppBundle\Entity\Page
            translation:
                classes:
                    model: AppBundle\Entity\PageTranslation

```
```
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
    <entity name="AppBundle\Entity\PageTranslation" table="bitbag_cms_page_translation">
        <field name="subtitle" type="string" length="255" nullable="true"/>
    </entity>
</doctrine-mapping>
```

```
<?php

declare(strict_types=1);

namespace AppBundle\Entity;

use BitBag\SyliusCmsPlugin\Entity\PageTranslation as BasePageTranslation;

class PageTranslation extends BasePageTranslation
{
    /**
     * @var string|null
     */
    private $subtitle;

    /**
     * @return null|string
     */
    public function getSubtitle(): ?string
    {
        return $this->subtitle;
    }

    /**
     * @param null|string $subtitle
     */
    public function setSubtitle(?string $subtitle): void
    {
        $this->subtitle = $subtitle;
    }
}

```

this is error i am getting:

> An exception has been thrown during the rendering of a template ("An exception occurred while executing 'SELECT t0.id AS id_1, t0.slug AS slug_2, t0.name AS name_3, t0.meta_keywords AS meta_keywords_4, t0.meta_description AS meta_description_5, t0.content AS content_6, t0.subtitle AS subtitle_7, t0.locale AS locale_8, t9.type AS type_10, t9.path AS path_11, t9.id AS id_12, t9.owner_id AS owner_id_13, t0.translatable_id AS translatable_id_15 FROM bitbag_cms_page_translation t0 LEFT JOIN bitbag_cms_page_image t9 ON t9.owner_id = t14.id WHERE t0.translatable_id = ? AND t0.locale = ? LIMIT 1' with params [7, "hr_HR"]:
> 
> SQLSTATE[42S22]: Column not found: 1054 Unknown column 't14.id' in 'on clause'").

**Comment by antonioperic** [comment]
any update on this?

**Comment by bitbager** [comment]
Not yet, sorry

**Comment by fendrychl** [comment]
It can be easily resolved for entities that are defined in sylius_resource. It is necessary to define interface in sylius_resource + change targetEntity in ORM mapping to interface instead of class.

Is this solution ok for you @bitbager? I can send pull request.

Thanks.

**Comment by stefandoorn** [comment]
Does this also relate to #186 @fendrychl ?

**Comment by bitbager** [comment]
Closed due to #e051635c0f21865724807b43f0cbe650ed15c261.
